File: JavaScript_Courses/JavaScript_Basics/Preparation/06_Nested_Loops_Exercise/06_Cinema_Tickets.js
function cinemaTickets(input) { 
    let studentCurrent = 0;
    let stardardCurrent = 0;
    let kidCurrent = 0;
    
    let totalTickets = 0;
    
    
    let i = 0;
    let currentMovie = input[i];
    i++;
    
    while (currentMovie != 'Finish') {
    let seatAvailable = Number(input[i]);
    i++;
    let seatBougth = 0;
    
    while (true) {
    if (seatAvailable == seatBougth) {
    break;
    }
    
    let currentTicket = input[i];  
    i++;
    
    if (currentTicket == "End") {
    break;
    }
    
    seatBougth++;
    totalTickets++;
    if (currentTicket == "student") {
    studentCurrent++;
    } else if (currentTicket == "standard") {
    stardardCurrent++;
    } else if (currentTicket == "kid") {
    kidCurrent++;
    }
    
    }
    
    
    console.log(`${currentMovie} - ${(100 * seatBougth / seatAvailable).toFixed(2)}% full.`);
    
    
    currentMovie = input[i];
    i++;
    }
    
    
    console.log(`Total tickets: ${totalTickets}`);  
    console.log(`${(100 * studentCurrent / totalTickets).toFixed(2)}% student tickets.`);
    console.log(`${(100 * stardardCurrent / totalTickets).toFixed(2)}% standard tickets.`);
    console.log(`${(100 * kidCurrent / totalTickets).toFixed(2)}% kids tickets.`);
    
    }


// cinemaTickets([
//     "Taxi",
//     "10",
//     "standard",
//     "kid",
//     "student",
//     "student",
//     "standard",
//     "standard",
//     "End",
//     "Scary Movie",
//     "6",
//     "student",
//     "student",
//     "student",
//     "student",
//     "student",
//     "student",
//     "Finish",
// ]);

File: Python_Courses/Python_Basics/Preparation_PB_Exams/02_PB_Exam_6_7_April_2019/06_Cinema_Tickets.py
student_count = 0
standard_count = 0
kid_count = 0

command = input()
while not command == "Finish":
    current_movie = command
    all_seats = int(input())
    available_seats = all_seats
    current_count = 0

    current_ticket = input()
    while not current_ticket == "End":
        if current_ticket == "student":
            student_count += 1
        elif current_ticket == "standard":
            standard_count += 1
        elif current_ticket == "kid":
            kid_count += 1

        current_count += 1
        available_seats -= 1
        if available_seats <= 0:
            break

        current_ticket = input()

    print(f"{current_movie} - {current_count / all_seats * 100 :.2f}% full.")

    command = input()

total_tickets = student_count + standard_count + kid_count
print(f"Total tickets: {total_tickets}")
print(f"{student_count / total_tickets * 100 :.2f}% student tickets.")
print(f"{standard_count / total_tickets * 100 :.2f}% standard tickets.")
print(f"{kid_count / total_tickets * 100 :.2f}% kids tickets.")
